libwarp/solidity_prepass/WarpVisitor.cpp

```
@@ -498,7 +498,14 @@ void WarpVisitor::dynFuncArgsPass(const char* solFilepath) {
     this->setCompilerOptions(m_compiler);
     this->m_compiler->parse();
     this->m_compiler->analyze();
-    this->m_compiler->compile();
+
+    try{
+        this->m_compiler->compile();
+    } catch (boost::exception const& ex) {
+        auto error = this->m_compiler->errors().rbegin();
+        std::cout << "Compilation error : " << error->get()->what() << std::endl;
+        throw std::runtime_error("Compilation failed");
+    }
```

Why not rethrow the exception?
Why are we only printing the last error?
Why are we not logging any info from `ex`?

---

> Why not rethrow the exception?

Thanks. This has been implemented.

> Why are we only printing the last error?

I've now used a for loop which iterates only over the error that were generated during the corresponding compile command.

> Why are we not logging any info from `ex`?

I had tried this, and the only info that I could get from `ex` was a message saying `Compile called with errors`. I felt our error reporting made this redundant and hence I'm not logging anything from `ex`. Is there something more that I can use from `ex`? Please let me know, thanks!

---

If you rethrow, it's okay not to log the exception itself